### PR TITLE
add optional "assignVariable" attribute to "matchSpec"

### PR DESCRIPTION
**Pull request by gabrielleecredera**
- take "name" value and store in map with index "assignVariable" value
- e.g. you can check that an attribute exists, and assign the attribute value to a variable
- variables must start with "TFSEC_VAR_"
- the variable can then be used in any following matchSpec "value" attributes
- added the above to the custom check docs

solves #1337

### CODE REVIEW

**Comment by gabrielleecredera** [comment]
(not sure how to address the cyclomatic complexity issue, mainly caused by the switch)

**Review by owenrumney — APPROVED** [review]

**Comment by owenrumney** [comment]
> (not sure how to address the cyclomatic complexity issue, mainly caused by the switch)

I'll take a look at this shortly - thanks for the contribution :+1:

File: internal/app/tfsec/custom/custom_context.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+type customContext struct {
+	module    block.Module
+	variables customCheckVariables
+}
+
+func NewEmptyCustomContext() *customContext {
+	return &customContext{
+		module:    nil,
+		variables: make(map[string]string),
```
Comment (liamg) [inline internal/app/tfsec/custom/custom_context.go:15]: Nitpicking but should probably be `make(customCheckVariables)`

File: internal/app/tfsec/custom/custom_context.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	}
+}
+
+func NewCustomContext(module block.Module) *customContext {
+	return &customContext{
+		module:    module,
+		variables: make(map[string]string),
```
Comment (liamg) [inline internal/app/tfsec/custom/custom_context.go:22]: Nitpicking but should probably be `make(customCheckVariables)`

**Review by liamg — CHANGES_REQUESTED** [review]
Looks good, one minor nitpick

**Review by liamg — APPROVED** [review]

**Pull request merged**